pkg/virtualization/core/storage/service/ImageManagementService.go

```
@@ -193,38 +195,74 @@ func (ims *ImageManagementService) GetVirtualHardDiskConfig(path string) (size u
 	outparams := wmi.WmiMethodParamCollection{}
 	outparams = append(outparams, wmi.NewWmiMethodParam("SettingData", nil))
 	outparams = append(outparams, wmi.NewWmiMethodParam("Job", nil))
+	retryCount := 0
 
-	result, err := method.Execute(inparams, outparams)
-	if err != nil {
-		return
-	}
+	for {
```

Can you explain this scenario - we have been using this for many years, without hitting this kind of issue.

---

Seems to be very specific to Migrate scenario - can you confirm

---

There is no guarantee that the file will close in time. It is better to return this error or upper stack to retry.
Would like to keep the WMI code simple without these kind of logic.

---

This is not specific to migrate scenario. After retry, failure will still return to upper stack, having the retry in wmi would remove the need for every upper stack to retry on this kind of failure, this is a reliability improvement.